tests/test_factors.rs

```diff
@@ -0,0 +1,41 @@
+#[cfg(test)]
+mod tests {
+    use nalgebra as na;
+    use tiny_solver::factors::*;
+
+    #[test]
+    fn prior_factor() {
+        let prior_factor = PriorFactor {
+            v: na::dvector![3.0, 1.0],
+        };
+
+        let params = na::dvector![1.0, 2.0];
+
+        let residual = prior_factor.residual_func(&[params]);
+        assert_eq!(residual[0], -2.0);
+        assert_eq!(residual[1], 1.0);
+        let dual_params = na::dvector![
+            num_dual::DualDVec64::new(1.0, num_dual::Derivative::some(na::DVector::identity(1))),
+            num_dual::DualDVec64::new(2.0, num_dual::Derivative::some(na::DVector::identity(1)))
```

Comment:
I know this part is hard to read.
https://github.com/powei-lin/tiny-solver-rs/blob/98553af4f36cb1f4658118d72e1b94cef9f7a9db/src/residual_block.rs#L56